types/intercom-web/index.d.ts

```diff
@@ -46,6 +47,9 @@ declare namespace Intercom_ {
       website?: string;
       industry?: string;
     };
+
+    // Custom attributes
+    [custom_attribute: string]: any;
```

Comment:
```suggestion
    [custom_attribute: string]: string | number | boolean | null;
```
you can even create custom type to desscribe this (this is how it is doen for JSON schema types, so:
```ts
type CustomSchemaType =
    | string
    | number
    | boolean
    | null;
```

Reply:
Thanks for the input @peterblazejewicz. This was the approach I took at first but couldn't quite get Typescript to cooperate.

When `custom_attribute` is limited to those types, it expects all preceding entries in the `IntercomSettings` interface to match too. So when incorporating your suggestion, something like this pops up for the other attributes:
```
Property 'app_id' of type 'string | undefined' is not assignable to string index type 'string | number | boolean | null'.
```

I then thought the solution would be to add ` | undefined` to the `custom_attribute` types, but we have the same problem... just for the `company` object:
```
Property 'company' of type '{ id: string | number; name: string; created_at?: number | undefined; plan?: string | undefined; monthly_spend?: number | undefined; user_count?: number | undefined; size?: number | undefined; website?: string | undefined; industry?: string | undefined; } | undefined' is not assignable to string index type 'string | number | boolean | null | undefined'.
```

So that's why I stuck with the non-ideal `any` type. I've run into this issue with Typescript before but haven't ever found a more elegant solution than `any`.

Ideas?